GTFO-API/API/AssetAPI.cs

```diff
@@ -122,17 +135,39 @@ public static void LoadAndRegisterAssetBundle(byte[] bundleBytes)
         /// <param name="assetName">The original asset name</param>
         /// <param name="copyName">The name it should be cloned into</param>
         /// <returns>The <see cref="UnityEngine.Object"/> of the asset requested or null if it's not loaded</returns>
-        /// <exception cref="ArgumentException">The name is already registered</exception>
+        /// <exception cref="ArgumentException">The name you're trying to copy into is already registered</exception>
         public static UnityEngine.Object CloneAsset(string assetName, string copyName)
         {
             var originalAsset = GetLoadedAsset(assetName);
             if (originalAsset == null) throw new ArgumentException($"Couldn't find an asset with the name '{assetName}'", nameof(assetName));
+            if (IsAssetRegistered(copyName)) throw new ArgumentException($"The asset you're trying to copy into is already registered", nameof(copyName));
 
             var newAsset = UnityEngine.Object.Instantiate(originalAsset);
             RegisterAsset(copyName, newAsset);
             return GetLoadedAsset(copyName);
         }
 
+        /// <summary>
+        /// Clones an asset into a new one with a different name
+        /// </summary>
+        /// <typeparam name="T">Type of asset to extract</typeparam>
+        /// <param name="assetName">The original asset name</param>
+        /// <param name="copyName">The name it should be cloned into</param>
+        /// <returns>The asset requested as <typeparamref name="T"/> or null if it's not loaded</returns>
+        /// <exception cref="ArgumentException">The name you're trying to copy into is already registered</exception>
+        /// <exception cref="InvalidCastException">The asset cannot be cast to <typeparamref name="T"/></exception>
+        public static T CloneAsset<T>(string assetName, string copyName) where T : UnityEngine.Object => CloneAsset(assetName, copyName)?.Cast<T>();
+
+        /// <summary>
+        /// Attempts to clone an asset into a new one with a different name
+        /// </summary>
+        /// <typeparam name="T">Type of asset to extract</typeparam>
+        /// <param name="assetName">The original asset name</param>
+        /// <param name="copyName">The name it should be cloned into</param>
+        /// <returns>The asset requested as <typeparamref name="T"/> or null if it's not loaded or cannot be cast</returns>
+        /// <exception cref="ArgumentException">The name you're trying to copy into is already registered</exception>
+        public static T TryCloneAsset<T>(string assetName, string copyName) where T : UnityEngine.Object => CloneAsset(assetName, copyName)?.TryCast<T>();
```

Comment:
Consider naming this to `TryInstantiate`. Also instead of returning the object utilize a bool return value and out parameter for consistency with most standard Try methods.